osu.Game/Graphics/UserInterface/RoundedSlider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osuTK;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public partial class RoundedSlider<T> : OsuSliderBar<T>
```

Comment:
I would avoid moving a known class implementation to a whole other name when possible, can't this be kept in `OsuSliderBar` and let `ShearedSlider` override any of the old design behaviour (similar to `OsuButton` and `RoundedButton`)?

Also I find it better to append `Bar` to the name to distinguish from osu! sliders, especially since this class is public and accessible anywhere.

Reply:
i mainly went with this split approach because i felt it was cleaner than adding a bunch of overrides to the normal slider, but i can do that if its preferred no problem 

And yeah good point about the names

Reply:
For what it's worth I think our button inheritance hierarchy is an uncontained trash fire and shouldn't be a model for anything. We've had multiple instances of changes in base classes ending up in visual regressions all over the place.

For this reason I'd argue for more separation between these implementations and not more. If there are common pieces to be extracted (like the sound playback or whatever) then they should be shipped out to separate classes and reused via composition IMO.

Reply:
Correct me if I'm wrong but despite both being inheritance isn't my current implementation less coupled to the normal slider than it would be if I'd implemented it though a bunch of overrides ?  Unless you were referring to something else. Sorry if I misunderstood . Eitherway I'll let you guys decide the best path forwards since I'm getting conflicting messages 😅

Reply:
I think the direction in this pull is not ideal, either. Example: `Nub` having pretty much its entire visual presentation replaced in the sheared variant. Put it this way: If I were to implement this, I'd probably inherit the framework-side `SliderBar` directly and leave `OsuSliderBar` well alone. Behaviours shared by the two (samples, etc.) I'd probably split to common components and add them *inside* implementations rather than faff about with inheritance.

Reply:
sounds easy enough, ill get on it